File: Linear_search/C++/linear_search.cpp
#include <iostream>
#include <vector>

int LinearSearch(const std::vector<int>& arr, int target) {
    for (int i = 0; i < arr.size(); i++) {
        if (arr[i] == target){
            return i;
        }
    }
    return -1;
}

int main() {
    std::vector<int> num = {123, 124, 163, 6, 4146};
    int required = 4146;
    int res = LinearSearch(num, required);
    if (res!= -1) {
        std::cout << "The element " << required << "found in position" << res << std::endl;
    } else {
            std::cout << "The element " << required << "not found" << std::endl;
    }
    return 0;
}

File: Linear_search/Java/linear_search.java
class LinearSearch {
    public static int linearSearch(int[] arr, int target) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == target) {
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args) {
        int[] num = {152, 7182, 976, 12, 5};
        int required = 5;

        int res = linearSearch(num, required);

        if (res != -1) {
            System.out.println("The element " + required + " found at position" + res);
        } else {
            System.out.println("The element " + required + " not found");
        }
    }
}